File: src/TestApp/TestRequestConsumer.cs

using MyServiceBus;
using Microsoft.Extensions.Logging;

namespace TestApp;

class TestRequestConsumer :
    IConsumer<TestRequest>
{
    private readonly ILogger<TestRequestConsumer> _logger;

    public TestRequestConsumer(ILogger<TestRequestConsumer> logger)
    {
        _logger = logger;
    }

    public async Task Consume(ConsumeContext<TestRequest> context)
    {
        var message = context.Message.Message;

        _logger.LogInformation("📨 Request: {Message}", message);

        if (DemoScenario.ShouldFaultRequest(message))
        {
            _logger.LogWarning("⚠️ TestRequest marked as fault case");
            throw new InvalidOperationException("TestRequest demo fault");
        }

        await context.RespondAsync(new TestResponse
        {
            Message = $"{message} 42"
        });
    }
}

File: src/Java/testapp/src/main/java/com/myservicebus/testapp/TestRequestConsumer.java
package com.myservicebus.testapp;

import java.util.concurrent.CompletableFuture;

import com.google.inject.Inject;
import com.myservicebus.ConsumeContext;
import com.myservicebus.Consumer;
import com.myservicebus.logging.Logger;
import com.myservicebus.logging.LoggerFactory;

class TestRequestConsumer implements Consumer<TestRequest> {
    private final Logger logger;

    @Inject
    public TestRequestConsumer(LoggerFactory loggerFactory) {
        this.logger = loggerFactory.create(TestRequestConsumer.class);
    }

    @Override
    public CompletableFuture<Void> consume(ConsumeContext<TestRequest> context) throws Exception {
        var message = context.getMessage().getMessage();

        logger.info("📨 Request: {}", message);

        if (DemoScenario.shouldFaultRequest(message)) {
            logger.warn("⚠️ TestRequest marked as fault case");
            throw new IllegalStateException("TestRequest demo fault");
        }

        var response = new TestResponse(message + " 42");
        return context.respond(response, context.getCancellationToken());
    }
}
